nixos/modules/services/web-servers/nginx/default.nix

```diff
@@ -580,7 +591,7 @@ in
     security.acme.certs = filterAttrs (n: v: v != {}) (
       let
         vhostsConfigs = mapAttrsToList (vhostName: vhostConfig: vhostConfig) virtualHosts;
-        acmeEnabledVhosts = filter (vhostConfig: vhostConfig.enableACME) vhostsConfigs;
+        acmeEnabledVhosts = filter (vhostConfig: vhostConfig.enableACME && vhostConfig.useACMEHost == null) vhostsConfigs;
```

Comment:
Didn't you mean to use `hostConfig.enableACME || vhostConfig.useACMEHost == null` here?

A few lines above you make them mutually exclusive thus they can never be set at the same time. The list would always be empty.